capstone-frontend/src/components/BookSearchTile.jsx

```diff
@@ -1,34 +1,87 @@
-import React from 'react';
-import { Container, Row, Col } from 'react-bootstrap';
+import React, { Component } from 'react';
+import { Button, Col, Row } from 'react-bootstrap';
+import StarRatings from 'react-star-ratings'
 import { BookListAddDropdown } from './BookListAddDropdown';
 import { BookDescriptionOverlay } from './BookDescriptionOverlay';
 import '../styles/Book.css';
 
-const BookSearchTile = (props) => {
-  return (
-    <div className='book-search-tile'>
-      <Container>
-        <Row className='justify-content-md-center'>
-          <Col md='auto'>
-            <BookDescriptionOverlay book={props.book}>
-              <img className='book-img-md' src={props.book.thumbnailLink} alt={props.book.title} />
-            </BookDescriptionOverlay>
-          </Col>
-          <Col>
-            <div className='center-vertical'>
-              <h2 className='book-title'> {props.book.title} </h2>
-              <p className='book-authors'> {props.book.authors.join(', ')} </p>
-            </div>
-          </Col>
-          <Col md='auto'>
-            <Container className='center-vertical'>
-              <BookListAddDropdown bookLists={props.bookLists} updateBookLists={props.updateBookLists} book={props.book} />
-            </Container>
-          </Col>
-        </Row>
-      </Container>
-    </div>
-  );
+class BookSearchTile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasRead: false,
+      bookObject: {},
+    }
+    fetch(`/api/books?userID=${window.localStorage.getItem('userID')}&gbookID=${this.props.book.id}`)
+      .then(response => response.json()).then(books => {
+        if (books.length === 1) {
+          this.setState({hasRead: books.length === 1, bookObject: books[0]});
+        }
+      })
+      .catch(e => console.log(e));
+  }
+
+  markRead = () => {
+    const readBook = {
+      gbookID: this.props.book.id,
+      userID: window.localStorage.getItem('userID'),
+    }
+    fetch('/api/books', {method:'post', body: JSON.stringify(readBook)})
+        .then(response => response.json())
+        .then(bookObject => this.setState({hasRead: true, bookObject}))
+        .catch(e => console.log(e));
```

Comment:
nit: These catches are console.logs instead of alerts like elsewhere in the application, should we move forward using console logs or alerts?

Reply:
I believe we should start using console logs, just because some of the error reporting is unhelpful and slows down user experience.